mvvmfx/src/main/java/de/saxsys/mvvmfx/internal/viewloader/ViewLoaderReflectionUtils.java

```diff
@@ -410,6 +413,16 @@ static Object injectScopeIntoField(Field scopeField, Object viewModel, ContextIm
         }
     }
 
+    private static <ViewModelType extends ViewModel> Method getInitializeMethod(ViewModelType viewModel) {
+        for (Method method : viewModel.getClass().getDeclaredMethods()) {
+            if(method.isAnnotationPresent(Initialize.class)) {
+                method.setAccessible(true);
```

Comment:
If we change the accessibility of the method we need to reset it after invokation. 
For this reason I would remove this line 409 here and instead change the invokation logic at line 405.
It has to look something like this:

```java
AccessController.doPrivileged((PrivilegedAction)() -> {
    boolean wasAccessible = initMethod.isAccessible();
    try {
        return initMethod.invoke(viewModel);
    } catch (InvocationTargetException | IllegalAccessException e) {
        throw ...
    } finally {
        initMethod.setAccessible(wasAccessible);
    }
});
```

For Fields we have convinient helper methods for this usecase in `ReflectionUtils`. For our usecase the method `public static <T> T accessField(final Field field, final Callable<T> callable, String errorMessage)` at line 109 in `ReflectionUtils` would be suitable. However, at the moment this method is targeted only for Fields and not for Methods. But actually we could change the Type of the first parameter to `java.lang.reflect.AccessibleObject` which includes both Field and Method and has everything we need.

I don't know if you are familiar with the reflection API and are able to do this refactoring? If not, no problem, then I would take over this task.